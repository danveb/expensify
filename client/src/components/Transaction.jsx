import { useContext } from "react";
import { GlobalContext } from "../context/GlobalState";

const Transaction = ({ transaction }) => {
    // useContext 
    const { deleteTransaction } = useContext(GlobalContext); 
    
    const sign = transaction.amount < 0 ? "-" : "+"; 

    // handleDeleteTransaction
    const handleDeleteTransaction = () => {
        deleteTransaction(transaction.id);
    };

    return (
        <li className={transaction.amount < 0 ? "minus" : "plus"}>
            {transaction.text} <span>{sign} ${Math.abs(transaction.amount)}</span><button className="delete-btn" onClick={handleDeleteTransaction}>X</button>
        </li>
    )
}

export default Transaction