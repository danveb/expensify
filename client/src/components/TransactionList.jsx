import { useContext } from "react";
import { GlobalContext } from "../context/GlobalState"; 
import Transaction from "./Transaction";

const TransactionList = () => {
    // useContext
    const context = useContext(GlobalContext); 
    // destructure transactions from context
    const { transactions } = context; 

    return (
        <>
            <h3>History</h3>
            <ul id="list" className="list">
                {/* map through transactions from contextAPI and pass thru props to Transaction component */}
                {transactions.map((transaction) => (
                    <Transaction key={transaction.id} transaction={transaction} />
                ))}
            </ul>
        </>
    )
}

export default TransactionList