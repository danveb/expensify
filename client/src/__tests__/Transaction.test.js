import { render } from "@testing-library/react";
import TransactionList from "../components/TransactionList"; 
import Transaction from "../components/Transaction"; 

describe("Transaction component", () => {
    // smoke test
    test("renders Transaction component", () => {
        render(<TransactionList>
            <Transaction />
        </TransactionList>); 
    });

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(<TransactionList>
                <Transaction />
            </TransactionList>); 
        expect(asFragment()).toMatchSnapshot(); 
    });
});

// needs work; jest failing