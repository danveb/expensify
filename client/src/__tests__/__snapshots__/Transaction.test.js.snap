// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component matches snapshot 1`] = `
<DocumentFragment>
  <h3>
    History
  </h3>
  <ul
    class="list"
    id="list"
  >
    <li
      class="minus"
    >
      Groceries 
      <span>
        - $200
      </span>
      <button
        class="delete-btn"
      >
        X
      </button>
    </li>
    <li
      class="minus"
    >
      Cooking Class 
      <span>
        - $150
      </span>
      <button
        class="delete-btn"
      >
        X
      </button>
    </li>
    <li
      class="plus"
    >
      Freelance 
      <span>
        + $7000
      </span>
      <button
        class="delete-btn"
      >
        X
      </button>
    </li>
    <li
      class="minus"
    >
      Macbook Air 
      <span>
        - $1700
      </span>
      <button
        class="delete-btn"
      >
        X
      </button>
    </li>
    <li
      class="minus"
    >
      Rent 
      <span>
        - $2600
      </span>
      <button
        class="delete-btn"
      >
        X
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
